tools/topology/topology2/doc/topology2-filter.py

```diff
@@ -0,0 +1,747 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2023 Intel Corporation. All rights reserved.
+#
+# The usage of this command is simple. It takes one argument, a path
+# to a SOF topology2 source file, and it produces to stdout something
+# that can be parsed by Doxygen as C source. Produced output contains
+# C struct definitions and instantiations analogous to the topology2
+# classes and object instances.
+#
+# The purpose of the translated C code is not to document actual
+# topology2 code, but only to provide anchors for Doxygen to form a
+# network of links through which to navigate the topology sources and
+# find the pieces of related Doxygen documentation. The filter also
+# creates separate pages of the original code and adds links next to the
+# pages in the C struct definition, instance documentation and their
+# possible Doxygen documentation.
+
+import sys
+import os
+import re
+import io
+
+debug = open("filter_debug.txt", "a", encoding="ascii")
```

Comment:
```suggestion
logging.basicConfig(filename="filter_debug.txt",...
```
https://docs.python.org/3/howto/logging.html

(and no this is not "overkill", this module is standard, built-in, fast and dead simple to use)